pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -124,11 +158,13 @@ func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.
 			for {
 				select {
 				case <-ticker.C:
-					err = p.loadConfiguration(routineCtx, configurationChan)
+					err = p.loadConfiguration(routineCtx, certInfo, configurationChan)
 					if err != nil {
 						return fmt.Errorf("failed to refresh consul catalog data: %w", err)
 					}
-
+				case certInfo = <-p.certChan:
+					// nothing much to do, next ticker cycle will propagate
```

Comment:
Good point, I will think about it. But usually the `RefreshInterval` should be way smaller than the cert lifetime (seconds vs days). But to prevent errors it probably makes sense to immediately build it